src/change_stream.ts

```diff
@@ -220,22 +220,23 @@ export class ChangeStream extends EventEmitter {
 
     this.parent = parent;
     this.namespace = parent.s.namespace;
+
     if (parent instanceof Collection) {
       this.type = CHANGE_DOMAIN_TYPES.COLLECTION;
-      this.topology = parent.s.db.s.topology;
     } else if (parent instanceof Db) {
       this.type = CHANGE_DOMAIN_TYPES.DATABASE;
-      this.topology = parent.s.topology;
     } else if (parent instanceof MongoClient) {
       this.type = CHANGE_DOMAIN_TYPES.CLUSTER;
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      this.topology = parent.topology!;
     } else {
       throw new TypeError(
         'parent provided to ChangeStream constructor is not an instance of Collection, Db, or MongoClient'
       );
     }
 
+    const topology = getTopology(parent);
+    if (!topology) throw new MongoClientClosedError();
+    this.topology = topology;
```

Comment:
we should be able to use `getTopology` to remove the need to attach a `topology` property on `ChangeStream` as well